spec/integration/apply_spec.rb

```diff
@@ -4,266 +4,288 @@
 require 'bolt_spec/conn'
 require 'bolt_spec/files'
 require 'bolt_spec/integration'
+require 'bolt_spec/project'
 require 'bolt_spec/puppet_agent'
 require 'bolt_spec/run'
 
-describe "apply", expensive: true do
+TEST_VERSIONS = [
+  [5, 'puppet5'],
+  [6, 'puppet6'],
+  [7, 'puppet']
+].freeze
+
+describe 'apply', expensive: true do
   include BoltSpec::Conn
-  include BoltSpec::Files
   include BoltSpec::Integration
+  include BoltSpec::Project
   include BoltSpec::PuppetAgent
   include BoltSpec::Run
 
-  let(:modulepath) { File.join(__dir__, '../fixtures/apply') }
-  let(:hiera_config) { File.join(__dir__, '../fixtures/configs/empty.yml') }
-  let(:config_flags) { %W[--format json --targets #{uri} --password #{password} --modulepath #{modulepath}] + tflags }
-  let(:apply_settings) { { 'show_diff' => true } }
+  let(:project_config) do
+    {
+      'apply_settings' => { 'show_diff' => true },
+      'hiera-config'   => File.join(__dir__, '../fixtures/configs/empty.yml'),
+      'modulepath'     => File.join(__dir__, '../fixtures/apply')
+    }
+  end
 
-  describe 'over ssh', ssh: true do
-    let(:uri) { conn_uri('ssh') }
-    let(:user) { conn_info('winrm')[:user] }
-    let(:password) { conn_info('ssh')[:password] }
-    let(:tflags) { %W[--no-host-key-check --run-as root --sudo-password #{password}] }
+  # The following are run with both *nix and Windows targets.
+  shared_examples 'agentful tests' do |targets|
```

Comment:
It seems like we lost the `does not create project` test?

Reply:
I actually meant to ask about that test... I'm not really sure why we have that test to begin with or if it's providing any value. As far as I'm aware, installing the agent will never create `~/.puppetlabs` on a target.

Reply:
Hm I think I'm ok with that, since I'd be surprised if it also made a bolt directory there.